peachjam/models/core_document_model.py

```diff
@@ -896,6 +896,87 @@ def save(self, *args, **kwargs):
             # first save, set the download filename
             self.set_download_filename()
 
+    def delete(self, *args, **kwargs):
+        if (
+            hasattr(self.document, "publication_file")
+            and self.document.publication_file.use_source_file
+        ):
+            raise ValidationError(
```

Comment:
I don't think we should do this, because it could potentially break general deletion of a document, such as during ingestion.